### PR TITLE
made stricter checking of residue types in sequence

### PR DESCRIPTION
**Pull request by dcchivian**

### CODE REVIEW

**Comment by dcchivian** [comment]
psiBLAST App was letting nucleotide-only MSAs through and failing during execution.  Added stricter tests throughout BLAST Apps that require protein input.  Unit tests are passing.

**Comment by dcchivian** [comment]
@sychan @psdehal @scanon @JamesJeffryes addresses PTV-1077

File: lib/kb_blast/kb_blastImpl.py [hunk 1]
```diff
@@ -1861,8 +1861,10 @@ def BLASTp_Search(self, ctx, params):
             sequence_str = input_one_data['sequences'][0]['sequence']
 
             PROT_pattern = re.compile("^[acdefghiklmnpqrstvwyACDEFGHIKLMNPQRSTVWYxX ]+$")
-            #DNA_pattern = re.compile("^[acgtuACGTUnryNRY ]+$")
-            if not PROT_pattern.match(sequence_str):
+            DNA_pattern = re.compile("^[acgtuACGTUnryNRY ]+$")
+            if DNA_pattern.match(sequence_str):
+                self.log(invalid_msgs,"Need protein query sequence.  BAD nucleotide record for sequence_id: "+header_id+"\n"+sequence_str+"\n")
```
Comment (JamesJeffryes) [inline lib/kb_blast/kb_blastImpl.py]: This line is too long, please break into multiple lines

**Review by JamesJeffryes — CHANGES_REQUESTED** [review]
The logic looks fine but I think it would be pretty simple and valuable to add unit tests that make sure the right error is raised for these cases where nucleotides are passed to these functions

**Comment by dcchivian** [comment]
@JamesJeffryes I made your suggested changes.  Please review.  Thanks!

**Comment by dcchivian** [comment]
unit tests (including new one with nucleotide MSA for PSI-BLAST) are all passing

**Review by JamesJeffryes — APPROVED** [review]
I can confirm the tests check out. I'm ok with this PR

**Pull request merged**